### DISCUSSION TITLE
AbuseIPDB integration?

### DISCUSSION THREAD

**Discussion by karolyi** [wdes/security#223] [Ideas]
Hey,

great list of banned addresses/ranges, I did something similar on my (FreeBSD) server but seeing yours, I just started to use this. Thanks!

One extra that isn't to be found here is, an AbuseIPDB downloadable range: it gives you 10000 hosts their honeypots and contributors — like me —  in the free version.

I already have a python script running for it, are you interested in integrating that as well?

Here's the project source: https://gitea.ksol.io/karolyi/abuseipdb-fetcher

**Comment by williamdes** [comment]
Hello !

Thank you for reaching out to me. A new format for AbuseIPDB would be welcome.
Can the python script be a bash script ?

Will I need to submit the list to AbuseIPDB ?

**Comment by williamdes** [comment]
You may want to read https://github.com/datacenters-network/security/discussions/224

**Comment by karolyi** [comment]
I'm an oldschool python guy so I used python for the parser. I think they also offer a simple text output format (mine parses the JSON output) but for that you'd have to consult their API documentation.

You don't need to submit anything, I've just set up my fail2ban to do so, helping maintaining the database in doing so.

The only downside is, in order to download even the smallest list (that only consists 10.000 records) you'll need to register and get an API key. Paid subscriptions get more addresses.

I'll express my opinions in the other ticket.

**Reply by williamdes** [reply]
I really do not understand this request. Can you help me?

Do I have to implement something?
Or should I report to the AbuseIP API the IPs I find?

**Reply by karolyi** [reply]
You don't need to implement — or submit — anything if you don't want to :)

My project is free to be used, you can run it once you collect the IPs from your scheduled scripts, and add the result to your artifacts (that is, the files you generate).

The only necessary thing for using abuseipdb is a registration, that will give you the API token to download their ip list.
